cluster-autoscaler/cloudprovider/clusterapi/clusterapi_provider.go

```diff
@@ -68,8 +70,14 @@ func (p *provider) nodeNames(machineSet *v1alpha1.MachineSet) ([]string, error)
 			glog.Errorf("Status.NodeRef of machine %q does not reference a node (rather %q)", machine.Name, machine.Status.NodeRef.Kind)
 			continue
 		}
-		if machineOwnerName(machine) == machineSet.Name {
-			nodeNames = append(nodeNames, machine.Status.NodeRef.Name)
+
+		node, err := p.machineController.findNodeByNodeName(machine.Status.NodeRef.Name)
+		if err != nil {
+			return nil, fmt.Errorf("unknown node %q", machine.Status.NodeRef.Name)
+		}
+
+		if node != nil {
+			nodeNames = append(nodeNames, node.Spec.ProviderID)
```

Comment:
Why do you append the ProviderID into the list of node names?

Reply:
The variable name is badly named now.

Reply:
Renamed.